app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt

```diff
@@ -18,141 +19,96 @@ class FirebaseGameLobbyRepository : IGameLobbyRepository {
-        val gameID = abs(gameName.hashCode()) //TODO: Better id generation
-        fs.collection("games").add(
-                Game(
-                        gameID,
-                        gameName,
-                        LocalUser.pseudo,
-                        gameDuration,
-                        emptyMap(), //TODO: For now no params
-                        GameState.LOBBY,
-                        emptyList(), //TODO: Add local user participation
-                        Date(),
-                        Date((Int.MAX_VALUE / 2).toLong()), //TODO: For now the game is available to the max
-                        Date(),
-                        IDHelper.getPlayerID()
-                )
-        )
-        return gameID
+    override fun createGame(gameName: String, gameDuration: Long): String {
+        val newGameRef = fs.collection(GAME_COLLECTION).document() //TODO: Better id generation
+        newGameRef.set(Game(
+                newGameRef.id,
+                gameName,
+                LocalUser.uid,
+                gameDuration,
+                emptyMap(), //TODO: For now no params
+                listOf(Participation(LocalUser.uid, Faction.PREDATOR, false, "", "")), //TODO: Add local user participation
+                Date(),
+                Date((Int.MAX_VALUE / 2).toLong()), //TODO: For now the game is available to the max
+                Date(),
+                GameState.LOBBY
+        ))
+        return newGameRef.id
     }
 
-    override fun getGameName(gameId: Int, cb: Callback<String>) {
-        getFirebaseGameWithID(gameId) { fid ->
-            fs.collection("games").document(fid).get().addOnSuccessListener {
-                cb(it["name"] as String)
-            }
+    override fun getGameName(gameId: String, cb: Callback<String>) {
+        fs.collection(GAME_COLLECTION).document(gameId).get().addOnSuccessListener {
+            cb(it["name"] as String)
         }
     }
 
-    override fun getGameDuration(gameId: Int, cb: Callback<Long>) {
-        getFirebaseGameWithID(gameId) { fid ->
-            fs.collection("games").document(fid).get().addOnSuccessListener {
-                cb(it["duration"] as Long)
-            }
+    override fun getGameDuration(gameId: String, cb: Callback<Long>) {
+        fs.collection(GAME_COLLECTION).document(gameId).get().addOnSuccessListener {
+            cb(it["duration"] as Long)
         }
     }
 
-    override fun getPlayers(gameId: Int, cb: Callback<List<Player>>) {
+    override fun getPlayers(gameId: String, cb: Callback<List<Player>>) {
         getParticipations(gameId) {
-            cb(it.map { x ->
-                //callback to the right instance of Player
-                if(x.faction == PlayerFaction.PREY)
-                    Prey(x.playerID, x.tag)
-                else
-                    Predator(x.playerID)
-            })
+            cb(it.withIndex().map { x -> x.value.toPlayer(x.index) })
```

Comment:
Are we sure that firebase always returns the list in the same order ? Because we can't afford to have different player ids

Reply:
I agree with your point. Even if firebase does so, it might be good to order them in a deterministic way independent of the backend.
I suggest we always sort the player by userID before taking the index